pubsub_logging/utils.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities for the Python logging handlers."""
+
+
+import base64
+import sys
+import threading
+
+from googleapiclient import discovery
+from googleapiclient import errors
+
+import httplib2
+
+from oauth2client.client import GoogleCredentials
+
+from pubsub_logging.errors import RecoverableError
+
+
+PUBSUB_SCOPES = ["https://www.googleapis.com/auth/pubsub"]
+
+clients = threading.local()
```

Comment:
These handlers can be shared with several threads in users' code. PubsubHandler don't spawn child process, so those threads will share the same client if I remove this part, which is problematic.